OpenRA.Mods.Common/Traits/DamagedByTerrain.cs

```diff
@@ -9,17 +9,24 @@
  */
 #endregion
 
+using System.Collections.Generic;
 using System.Linq;
 using OpenRA.GameRules;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("This actor receives damage from the given weapon when on the specified terrain type.")]
-	class DamagedByTerrainInfo : UpgradableTraitInfo, IRulesetLoaded, Requires<HealthInfo>
+	class DamagedByTerrainInfo : UpgradableTraitInfo, Requires<HealthInfo>
 	{
-		[Desc("The weapon which is used to damage the actor.")]
-		[WeaponReference, FieldLoader.Require] public readonly string Weapon;
+		[Desc("The damage the actor suffers on these terrain.")]
```

Comment:
I don't think there is any need to be repetitive here, the user knows this relates to the given terrain types.
`Amount of damage received per tick.` would be more helpful.